server/api/session/manager.go

```diff
@@ -0,0 +1,41 @@
+package session
+
+import (
+	"crypto/rand"
+	"sync"
+	"time"
+)
+
+const CookieName = "id"
+const ExpiresIn = 2 * time.Hour
+
+type Manager interface {
+	NewSession() (string, error)
+}
+
+
+type inMemoryManager struct {
+	sessions map[string]time.Time
+	mutex *sync.Mutex
+}
+
+func NewInMemoryManager() Manager {
+	return &inMemoryManager{
+		sessions: make(map[string]time.Time),
+		mutex: &sync.Mutex{},
+	}
+}
+
+func (i inMemoryManager) NewSession() (string, error)  {
+	b := make([]byte, 32)
+	_, err := rand.Read(b)
+	if err != nil {
+		return "", err
+	}
+
+	sessionId := string(b)
```

Comment:
I would encode this right away and not store it as bytes in your map. It looks like you are doing base64 encoding before returning this to the client anyway, I would just do that here.